[CssBaseline] Make dark mode scrollbar overrides an optional function (#24780)
mui-org/material-ui: docs/src/pages/components/css-baseline/css-baseline.md

### Scrollbars

The colors of the scrollbars can be customized to improve the contrast (especially on Windows). Add this code to your theme (for dark mode).

```jsx
import darkScrollbar from '@material-ui/core/darkScrollbar';

const theme = createMuiTheme({
  components: {
    MuiCssBaseline: {
      styleOverrides: {
        body: theme.palette.mode === 'dark' ? darkScrollbar() : null,
      },
    },
  },
});
```

> The documentation website you are on uses `darkScrollbar` when the dark mode is enabled.

Reviewer: ```suggestion
This website uses `darkScrollbar` when dark mode is enabled.
```

> However, be aware that using this help (and customizing `-webkit-scrollbar`) forces macOS to always show the scrollbar.

Reviewer: ```suggestion
Be aware, however, that using this utility (and customizing `-webkit-scrollbar`) forces MacOS to always show the scrollbar.
```
